### PR TITLE
Avoid ball getting stuck in paddle

### PR DESCRIPTION
**Pull request by naomijub**

### CODE REVIEW

File: src/state/mod.rs [hunk 1]
```diff
@@ -21,7 +22,7 @@ pub struct MainState {
 impl MainState {
     pub fn new() -> GameResult<MainState> {
         let s = MainState {
-            ball: Ball::new(400.0, 300.0),
+            ball: Ball::new(WINDOW_WIDTH / 2.0, WINDOW_HEIGHT / 2.0),
```
Comment (evaporei) [inline src/state/mod.rs:25]: Nice that you've fixed this!

File: src/ball/mod.rs [hunk 2]
```diff
@@ -53,9 +53,11 @@ impl Ball {
         let ball = self.position;
 
         players.iter().map(|p| p.paddle_vertex()).for_each(|p| {
-            if p.chunks(2)
-                .any(|c| dist_to_segment(ball, c[0], c[1]) < BALL_RADIUS)
-            {
+            if p.chunks(2).any(|c| {
+                dist_to_segment(ball, c[0], c[1]) < BALL_RADIUS
+                    && ((self.position[0] < WINDOW_WIDTH / 2.0 && self.direction[0] < 0.0)
```
Comment (evaporei) [inline src/ball/mod.rs]: What do you think of adding a variable for these checks.

Something like:

```rust
let is_ball_vertial_stuck = (self.position[0] < WINDOW_WIDTH / 2.0 && self.direction[0] < 0.0)
            || (self.position[0] > WINDOW_WIDTH / 2.0 && self.direction[0] > 0.0);
if p.chunks(2).any(|c| {
    dist_to_segment(ball, c[0], c[1]) < BALL_RADIUS
        && is_ball_vertial_stuck
}) {
    self.direction[0] = -self.direction[0];
}
```

The name is just an example, if you have something better go for it :slightly_smiling_face:

**Comment by evaporei** [comment]
Also, I've tested locally and it works, but I prefer that the sides of the paddle would also make the ball change direction.
We could do this in another PR though, and idk if you would agree on this design decision. I just think it is a better experience.

File: src/ball/mod.rs [hunk 3]
```diff
@@ -84,6 +85,23 @@ impl Ball {
     }
 }
 
+fn is_ball_in_correct_direction(x_pos: f32, x_dir: f32) -> bool {
```
Comment (evaporei) [inline src/ball/mod.rs:88]: I'm a bit confused, what does it mean `correct direction`?

Comment (naomijub) [reply src/ball/mod.rs:88]: assures that the ball is in the correct direction for the quadrant, this is how we prevet it from being stuck

**Review by evaporei — APPROVED** [review]

**Comment by naomijub** [comment]
@otaviopace tu foi mais rapido dessa vez!!! 😡

**Pull request merged**